cmd/autocompletion.go

```diff
@@ -0,0 +1,242 @@
+completion of aws-okta commands.
+
+Bash
+	$ source <(aws-okta completion bash)
+will load the aws-okta completion code for bash. Note that this depends on the
+bash-completion framework. It must be sourced before sourcing the aws-okta
+completion, e.g. on macOS:
+	$ brew install bash-completion
+	$ source $(brew --prefix)/etc/bash_completion
+	$ source <(aws-okta completion bash)
+	(or, if you want to preserve completion within new terminal sessions)
+	$ echo 'source <(aws-okta completion bash)' >> ~/.bashrc
+
+Zsh
+	$ source <(aws-okta completion zsh)
+	(or, if you want to preserve completion within new terminal sessions)
+	$ echo 'source <(aws-okta completion zsh)' >> ~/.zshrc`,
+}
+
+var bashAutocompleteCmd = &cobra.Command{
+	Use:   "bash",
+	Short: "Output shell completion code for bash",
+	Long: `
+Output shell completion code for bash.
+This command prints shell code which must be evaluated to provide interactive
+completion of aws-okta commands.
+	$ source <(aws-okta completion bash)
+will load the aws-okta completion code for bash. Note that this depends on the
+bash-completion framework. It must be sourced before sourcing the aws-okta
+completion, e.g. on macOS:
+	$ brew install bash-completion
+	$ source $(brew --prefix)/etc/bash_completion
+	$ source <(aws-okta completion bash)
+	(or, if you want to preserve completion within new terminal sessions)
+	$ echo 'source <(aws-okta completion bash)' >> ~/.bashrc`,
+	RunE: runCompletionBash,
+}
+
+var zshAutocompleteCmd = &cobra.Command{
+	Use:   "zsh",
+	Short: "List users",
+	Long: `
+Output shell completion code for zsh.
+This command prints shell code which must be evaluated to provide interactive
+completion of aws-okta commands.
+	$ source <(aws-okta completion zsh)
+	(or, if you want to preserve completion within new terminal sessions)
+	$ echo 'source <(aws-okta completion zsh)' >> ~/.zshrc
+zsh completions are only supported in versions of zsh >= 5.2`,
+	RunE: runCompletionZsh,
+}
+
+func runCompletionBash(cmd *cobra.Command, args []string) error {
+	out := bufio.NewWriter(os.Stdout)
+	defer out.Flush()
+	return RootCmd.GenBashCompletion(out)
+}
+
+// Copyright 2016 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+func runCompletionZsh(cmd *cobra.Command, args []string) error {
+	out := bufio.NewWriter(os.Stdout)
+	defer out.Flush()
+	zshInitialization := `
+__aws-okta_bash_source() {
```

Comment:
This is pretty ugly, but meh, shell code amirite?

Reply:
There's still a bug open for the Zsh completion https://github.com/spf13/cobra/issues/107